source/commands/bounty.js

```diff
@@ -0,0 +1,65 @@
+const { PermissionFlagsBits, ActionRowBuilder, StringSelectMenuBuilder } = require('discord.js');
+const { CommandWrapper } = require('../classes');
+const { database } = require('../../database');
+const { getNumberEmoji } = require('../helpers');
+
+const customId = "bounty";
+const options = [];
+const subcommands = [
+	{
+		name: "post",
+		description: "Post your own bounty (+1 XP)",
+		optionsInput: []
+	}
+];
+module.exports = new CommandWrapper(customId, "Bounties are user-created objectives for other server members to complete", PermissionFlagsBits.ViewChannel, false, true, 3000, options, subcommands,
+	(interaction) => {
+		switch (interaction.options.getSubcommand()) {
+			case subcommands[0].name: // Post
+				database.models.Hunter.findOne({ where: { userId: interaction.user.id, guildId: interaction.guildId } }).then(async hunter => {
+					if (!hunter) { //TODO use findOrCreate after double-checking associations (may not cascade from hunter to user correctly)
+						const user = await database.models.User.findByPk(interaction.user.id);
+						if (!user) {
+							await database.models.User.create({ id: interaction.user.id });
+						}
+						hunter = await database.models.Hunter.create({ userId: interaction.user.id, guildId: interaction.guildId, isRankEligible: !interaction.member.manageable });
+					}
```

Comment:
This code of make-hunter-if-doesn't-exist probably belongs in its own utility function, since it'd be also useful when adding completers, completing a bounty outright, or toasting. I'd say that belongs in the DAO-like layer somewhere, but that's beyond the scope of this.

Reply:
I agree! However, I think Sequelize's out-of-the-box `findOrCreate` is exactly what we're looking, so we won't need this in the final code anyways. Unfortunately, because Hunter implies a User and I don't think we have associations configured correctly, switching to `findOrCreate` is juuust out of scope for this PR.

Reply:
There's a potential issue with `findOrCreate` where we want a `User` model created before the `Hunter` one due to the data structure and relationship between them, but in most places that are relevant we only really *need* the `Hunter`. This is something that we'll want to track as we implement `findOrCreate`, potentially abstracting this interaction into a separate function due to that relationship tie.

Reply:
IIRC, setting up associations correctly will include "Make a User if trying to make a Hunter and it's missing."